File: Algorithms/JavaScript/max-depth-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 * 
 * @param {TreeNode} root
 * @return {number}
 */
 
var maxDepth = function(root) {
  if (root === null) return 0;
  var left = maxDepth(root.left);
  var right = maxDepth(root.right);
  return (left > right ) ? (left + 1) : (right + 1)
};

File: Algorithms/Ruby/max-depth-binary-tree.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} root
# @return {Integer}

def max_depth_binary_tree(root)
  return 0 unless root
  left = max_depth_binary_tree(root.left)
  right = max_depth_binary_tree(root.right)

  left > right ? left + 1 : right + 1
end
